streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java

```diff
@@ -214,6 +214,70 @@ public void testStateOneThreadDeadButRebalanceFinish() throws InterruptedExcepti
         Assert.assertEquals(KafkaStreams.State.NOT_RUNNING, globalStreams.state());
     }
 
+    @Test
+    public void testStateOneAllThreadDead() throws InterruptedException {
```

Comment:
nit: rename -> `instanceStateShouldTransiteToErrorStateIfAllThreadsAreDead`

Also, just throw `Exception` for simplicity.

Reply:
ack